.github/workflows/test.yml

```diff
@@ -0,0 +1,66 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  test-and-maybe-uberjar:
+    # Runs the JVM and babashka tests across all OSs and JDKs, and
+    # creates an uberjar when the OS and JDK matches the UBERJAR_OS
+    # and UBERJAR_JDK env variables respectively.
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        jdk: [8, 11, 17]
+    env:
+      UBERJAR_OS: 'ubuntu'
+      UBERJAR_JDK: 8
+
+    runs-on: ${{matrix.os}}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Prepare java ${{ matrix.jdk }}
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: ${{ matrix.jdk }}
+
+      - name: Install Clojure
+        uses: DeLaGuardo/setup-clojure@9.5
+        with:
+          bb: '0.10.163'
+          cli: '1.10.3.1013'
+          lein: '2.9.10'
+
+      - name: Cache clojure dependencies
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.m2/repository
+            ~/.gitlibs
+          key: cljdeps-${{ hashFiles('deps.edn', 'bb.edn', 'project.clj') }}
+          restore-keys: cljdeps-
+
+      - name: Run JVM tests
+        run: bb jvm-test
+
+      - name: Run babashka tests
+        run: bb babashka-test
+
+      - name: Create ubejar
```

Comment:
The uberjar part isn't part of "additional tests on Github actions". We will still use our current CI setup for uploading a jar. I think this PR also uploads jar files for -RELEASE I think, which we shouldn't do. You can comment this out for future usage though.

Reply:
Or wait, does this only upload to Github actions artifacts, not to Github releases? Then it's ok

Reply:
I was thinking that the uberjar is the result of successful testing thus these two should ideally go together under the same job. Otherwise we are running the testing part twice (once in GH as part of testing the matrix, and a second time in circleci for the uberjar creation).

Nevertheless, I will remove the uberjar from GH as suggested with the next PR (since this has already been merged). Not entirely sure how this additional uberjar upload interferes with the existing setup. Thanks

Reply:
It doesn't, changing this is not necessary. I thought it uploaded to Github releases, but this is only a build artifact. So all good.